src/Endpoint/Partition.php

```diff
@@ -0,0 +1,171 @@
+<?php
+namespace Aws\Endpoint;
+
+use InvalidArgumentException as Iae;
+
+/**
+ * @internal
+ */
+final class Partition implements \ArrayAccess
+{
+    private $definition;
+
+    public function __construct(array $definition)
+    {
+        foreach (['partition', 'regions', 'services', 'dnsSuffix'] as $key) {
+            if (!isset($definition[$key])) {
+                throw new Iae("Partition missing required $key field");
+            }
+        }
+
+        $this->definition = $definition;
+    }
+
+    public function toArray()
+    {
+        return $this->definition;
+    }
+
+    public function offsetGet($offset)
+    {
```

Comment:
These method signatures are defined by [`ArrayAccess`](http://php.net/manual/en/class.arrayaccess.php), so we're stuck with the names they have, unfortunately.